backend/src/routes/AllowlistEntry.ts

```diff
@@ -17,21 +20,46 @@ function AllowlistEntryRoutes(allowlistService: typeof AllowlistEntry) {
 
   router.post('/', requireAdmin, async (req, res) => {
     try {
-      const { phoneNumber, destinationCountry } = req.body;
+      let validatedReq;
+      try {
+        const bodySchema = z.object({
+          phoneNumber: z.string(),
+          destinationCountry: z.string(),
+        });
+        const body = bodySchema.parse(req.body);
+        validatedReq = { body };
+      } catch (error) {
+        logger.error(error);
+        return res.status(400).send(error);
```

Comment:
This validation looks solid, but there's a lot of duplication between defining the schema and returning the 400 + logging the error. Do you think we can refactor this to reduce the duplication and make it cleaner?

Reply:
Here's an example i was able to come up with:

```js
// helper function file
function validateRequest<T>(schema: z.Schema<T>, handler: (parsedReq: T, res: Response) => unknown) {
  return (req: Request, res: Response) => {
    try {
      const parsedReq = schema.parse(req);
      return handler(parsedReq, res);
    } catch (error) {
      res.status(400).send(error);
    }
  }
}
```

```js
// In the route file
const POST_SCHEMA = z.object({
  body: z.object({
    phoneNumber: z.string(),
    destinationCountry: z.string(),
  })
});

  router.post('/', requireAdmin, validateRequest(POST_SCHEMA, async (req, res) => {
    try {
      const { phoneNumber, destinationCountry } = req.body;
      const allowlistEntry = await allowlistService.createAllowlistEntry({
        phoneNumber,
        destinationCountry,
      });
      return res.status(200).json(allowlistEntry);
    } catch (e) {
      return handleServiceError(e, res);
    }
  }));


```

Reply:
This would mean that `phoneNumber` would be type `any` is it? I wanted to try using a helper function too, but could not figure out how to preserve the types, which is pretty nice to have directly linked to the validator which is what zod provides. Not sure if the reduction in repetition of code is required compared to keeping it straightforward.

Reply:
My concerns with not helping a helper are that duplication is error prone, and we have to introduce quite a few nested blocks + return statements to preserve the correct order of operations. It feels a little too error-prone at the moment. Let me push a PR with a better implementation.